src/fsharp/TastOps.fs

```diff
@@ -1807,11 +1805,47 @@ let rec isUnmanagedTy g ty =
                 true
             elif tycon.IsStructOrEnumTycon then
                 match tycon.TyparsNoRange with
-                | [] -> tycon.AllInstanceFieldsAsList |> List.forall (fun r -> isUnmanagedTy g r.rfield_type) 
-                | _ -> false // generic structs are never 
+                | [] -> tycon.AllInstanceFieldsAsList |> List.forall (fun r -> isUnmanagedTy g r.rfield_type)
+                | typars ->
+                    // Handle generic structs
+                    // REVIEW: This may not be the most optimal, but it's probably better than having to iterate over all type arguments for every field.
```

Comment:
You can remove the comment, the code is fine perf wise

Reply:
Ok. For what it was worth, I did do some perf testing on it. I think the number was 15ms if it was called 10 thousand times, which the only way for that to happen is to have 10 thousand written calls that test the unmanaged constraint.